packages/frontend/src/hooks/useNativeTokenMaxValue.ts

```diff
@@ -0,0 +1,243 @@
+import { useState, useCallback } from 'react'
+import { BigNumber } from 'ethers'
+import { useWeb3Context } from 'src/contexts/Web3Context'
+import logger from 'src/logger'
+import Transaction from 'src/models/Transaction'
+import { Token, TProvider } from '@hop-protocol/sdk'
+import { useApp } from 'src/contexts/AppContext'
+import Network from 'src/models/Network'
+import { ZERO_ADDRESS } from 'src/constants'
+
+export const METHOD_NAMES = {
+  convertTokens: 'convertTokens',
+  wrapToken: 'wrapToken',
+}
+
+const estimatedGasLimits = {
```

Comment:
I don't think the hard coded gas limits will be reliable because they are dynamic on rollups, since gas limits on ORUs are a function of current L1 gas prices. 

From [optimism docs](https://community.optimism.io/docs/developers/fees.html#transaction-fees-on-l2):

> You should always use the gas limit suggested by `eth_estimateGas`.

Is the idea that the hardcoded limits are only for estimates when a wallet is _not_ connected?